### PR TITLE
page cache: per-task-kind access stats

### PR DESCRIPTION
**Pull request by problame**
This PR adds a `task_kind` label to page cache access metrics.

These are to validate our hypothesis that the high hit page cache rate we observe in prod is due to internal tasks, not getpage requests from compute.
We believe the latter should near-always be a pageserver-page-cache _miss_ because compute has it's own page cache, and hence there is no locality of reference for its accesses to pageserver page cache.

Before this PR, we didn't have `RequestContext` propagation to any code below the on-demand downloader.
The vast majority of changes in this PR is concerned with adding that propagation.

### CODE REVIEW

File: pageserver/src/tenant/ephemeral_file.rs [hunk 1]
```diff
@@ -261,14 +275,24 @@ impl BlockReader for EphemeralFile {
...
     use crate::tenant::block_io::{BlockCursor, BlockReaderRef};
     use rand::{thread_rng, RngCore};
     use std::fs;
     use std::str::FromStr;
 
     fn harness(
         test_name: &str,
-    ) -> Result<(&'static PageServerConf, TenantId, TimelineId), io::Error> {
+    ) -> Result<
+        (
```
Comment (arpad-m) [inline pageserver/src/tenant/ephemeral_file.rs:288]: Maybe at this point it might make sense to introduce a `Harness` struct instead of returning tuples. But this can also be a followup change.

File: pgxn/neon/walproposer.c [hunk 2]
```diff
@@ -425,7 +425,7 @@ WalProposerPoll(void)
 
 			break;
 		}
-		
+
```
Comment (arpad-m) [inline pgxn/neon/walproposer.c:428]: These changes seem unrelated to me. can you remove them from this PR? Maybe put them into a separate one?

**Review by arpad-m — APPROVED** [review]

File: pageserver/src/tenant/ephemeral_file.rs (hunk 1 above)
Comment (problame) [reply pageserver/src/tenant/ephemeral_file.rs:288]: Yeah, not doing that here.
